src/App.js

```diff
@@ -76,11 +83,37 @@ const App = ({ setClient }) => {
   useGeolocationTracking()
   useCozyEnvironmentOverride()
 
+  const {
+    LauncherDialog,
+    canDisplayLauncher,
+    launcherClient,
+    launcherContext,
+    onKonnectorLog,
+    onKonnectorJobUpdate,
+    resetLauncherContext,
+    setLauncherContext
+  } = useLauncherContext()
+
   if (isLoading) {
     return null
   }
 
-  return <RootNavigator initialRoute={initialRoute} setClient={setClient} />
+  return (
+    <>
+      <RootNavigator initialRoute={initialRoute} setClient={setClient} />
+      {canDisplayLauncher() && (
+        <LauncherView
+          launcherClient={launcherClient}
+          launcherContext={launcherContext.value}
+          retry={resetLauncherContext}
+          setLauncherContext={setLauncherContext}
+          onKonnectorLog={onKonnectorLog}
+          onKonnectorJobUpdate={onKonnectorJobUpdate}
```

Comment:
Now that LauncherView is under LauncherContext, we can retrieve all those props directly in the component using `useLauncherContext`, so we don't need to inject them here.

Reply:
Yes but the LauncherView is a class component and I can't use hooks in it.

Reply:
And changing the LaucherView to a function would be more difficult at the moment.